src/node.cc

```diff
@@ -2849,66 +2853,127 @@ void ProcessEmitWarning(Environment* env, const char* fmt, ...) {
   f.As<v8::Function>()->Call(process, 1, &arg);
 }
 
+static bool PullFromCache(Environment* env,
+                          const FunctionCallbackInfo<Value>& args,
+                          Local<String> module,
+                          Local<Object> cache) {
+  Local<Object> exports;
+  Local<Context> context = env->context();
+  if (cache->Has(context, module).FromJust()) {
+    exports = cache->Get(context, module).ToLocalChecked()->
+        ToObject(context).ToLocalChecked();
+    args.GetReturnValue().Set(exports);
+    return true;
+  }
+  return false;
+}
+
+static Local<Object> InitModule(Environment* env,
+                                 node_module* mod,
+                                 Local<String> module) {
+  Local<Object> exports = Object::New(env->isolate());
+  // Internal bindings don't have a "module" object, only exports.
+  CHECK_EQ(mod->nm_register_func, nullptr);
+  CHECK_NE(mod->nm_context_register_func, nullptr);
+  Local<Value> unused = Undefined(env->isolate());
+  mod->nm_context_register_func(exports,
+                                unused,
+                                env->context(),
+                                mod->nm_priv);
+  return exports;
+}
 
-static void Binding(const FunctionCallbackInfo<Value>& args) {
-  Environment* env = Environment::GetCurrent(args);
+static void ThrowIfNoSuchModule(Environment* env, const char* module_v) {
+  char errmsg[1024];
+  snprintf(errmsg,
+           sizeof(errmsg),
+           "No such module: %s",
+           module_v);
+  env->ThrowError(errmsg);
+}
 
-  Local<String> module = args[0]->ToString(env->isolate());
-  node::Utf8Value module_v(env->isolate(), module);
+static void ThrowInvalidBindingName(Environment* env) {
+  env->ThrowError("Invalid binding name");
+}
 
+static void Binding(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  MaybeLocal<String> maybe_module = args[0]->ToString(env->context());
+  if (maybe_module.IsEmpty()) {
```

Comment:
In a similar vein, you can shorten this to:
```c++
Local<String> module;
if (!args[0]->ToString(env->context()).ToLocal(&module)) return;
```
There will be an exception pending.

Reply:
done